### PR TITLE
Add support for asymmetric padding in Conv and ConvTransposed.

### PR DESCRIPTION
**Pull request by Gurvan**
- Adds asymmetric padding to convolutions. The asymmetric padding is defined as a `Sequence` of `int` of length `2 * num_spatial_dim`, analog to `torch.nn.functional.pad`.

### CODE REVIEW

**Comment by patrick-kidger** [comment]
Nice, thank you. For asymmetric padding , can we make the API such that the user just directly passes e.g. `((1,2), (3,4), ...)` and we simply do `self.padding = padding`?

**Comment by Gurvan** [comment]
Oh I didn't see you already made the design choice in Pool.

Changes wrt the previous commit:
- made the `padding` API for Conv and ConvTranspose the same as the Pool one.
- added documentation about `padding` in the info section, rather than in the arguments description.
- added tests for padding with sequence of tuples for Conv and ConvTranspose

**Comment by patrick-kidger** [comment]
And merged! Thank you for contributing.

**Comment by Gurvan** [comment]
Thank you for taking the time!
